sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1798,14 +1837,58 @@ ProgramManager::kernelImplicitLocalArgPos(const std::string &KernelName) const {
   return {};
 }
 
+static bool shouldSkipEmptyImage(sycl_device_binary RawImg) {
```

Comment:
Hi, @steffenlarsen
As @jopperm pointed out, __sycl_register_lib may be called multiple times(e.g sycl dynamic library linking) and multiple user device images may all use bfloat16 device library, the previous logic will add duplicate bfloat16 device library image in exported_symbol map. I added check here to see whether native/fallback bfloat16 device library image has been handled, if yes, just skip it.
Could you help review again?
Thanks very much.